fdbclient/tests/mocks3_fixture.sh

```diff
@@ -0,0 +1,145 @@
+    BUILD_DIR="$build_dir_param"
+    if [ -z "$BUILD_DIR" ]; then
+        # Find the build directory - look for fdbserver binary
+        for dir in "../build_output" "../../build_output" "../../../build_output" "./build_output" "."; do
+            if [ -f "$dir/bin/fdbserver" ]; then
+                BUILD_DIR="$dir"
+                break
+            fi
+        done
+    fi
+
+    if [ -z "$BUILD_DIR" ]; then
+        echo "ERROR: Could not find fdbserver binary to run MockS3Server"
+        return 1
+    fi
+    
+    if [ ! -f "$BUILD_DIR/bin/fdbserver" ]; then
+        echo "ERROR: fdbserver binary not found at $BUILD_DIR/bin/fdbserver"
+        return 1
+    fi
+
+    # Try to start MockS3Server with port retry logic
+    for attempt in $(seq 1 $max_attempts); do
+        echo "Starting MockS3Server on ${MOCKS3_HOST}:${MOCKS3_PORT} (attempt $attempt/$max_attempts)"
+        
+        # Find an available port
+        if ! MOCKS3_PORT=$(find_available_port $MOCKS3_PORT); then
+            echo "ERROR: Could not find available port"
+            return 1
+        fi
+        echo "Using port: $MOCKS3_PORT"
+        
+        # Start MockS3Server using fdbserver with the new role
+        "$BUILD_DIR/bin/fdbserver" \
+            --role mocks3server \
+            --public-address "${MOCKS3_HOST}:${MOCKS3_PORT}" \
+            --listen-address "${MOCKS3_HOST}:${MOCKS3_PORT}" &
+        MOCKS3_PID=$!
+        
+        # Wait for server to be ready
+        local ready=false
+        for i in {1..10}; do
+            # Check if the process is still running
+            if ! kill -0 $MOCKS3_PID 2>/dev/null; then
+                echo "ERROR: MockS3Server process died"
+                break
+            fi
+            
+            # For now, just wait a bit since we're using a simulation
+            if [ $i -gt 3 ]; then
+                echo "MockS3Server ready"
+                ready=true
+                break
+            fi
+            sleep 1
+        done
+        
+        if [ "$ready" = true ]; then
+            return 0
+        fi
+        
+        # If we get here, the server failed to start properly
+        # Kill any remaining process and try next port
+        if kill -0 $MOCKS3_PID 2>/dev/null; then
+            kill $MOCKS3_PID 2>/dev/null || true
+            wait $MOCKS3_PID 2>/dev/null || true
+        fi
+        
+        # Try next port for next attempt
+        MOCKS3_PORT=$((MOCKS3_PORT + 1))
+    done
+    
+    echo "ERROR: MockS3Server failed to start after $max_attempts attempts"
+    return 1
+}
+
+shutdown_mocks3() {
+    if [ -n "$MOCKS3_PID" ]; then
+        echo "Shutting down MockS3Server (PID: $MOCKS3_PID)"
+        kill $MOCKS3_PID 2>/dev/null || true
+        wait $MOCKS3_PID 2>/dev/null || true
+    fi
+    
+}
+
+get_mocks3_url() {
+    echo "blobstore://mocks3:mocksecret@${MOCKS3_HOST}:${MOCKS3_PORT}"
+}
+
+# Provide compatibility functions with same names as seaweedfs_fixture.sh
```

Comment:
Do we need this? Is the intention to have AWS S3, seaweedFS and the MockS3 implementation as candidates for testing?